File: Spring-Boot/jpa-udemy/src/main/java/dev/vitorvidal/jpaudemy/entity/Review.java
package dev.vitorvidal.jpaudemy.entity;

import jakarta.persistence.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
public class Review {
    @Id
    @GeneratedValue
    private Long id;
    @Column
    private String rating;
    @Column
    private String description;
    @ManyToOne
    private Course course;
    @UpdateTimestamp
    private LocalDateTime lastUpdatedDate;
    @CreationTimestamp
    private LocalDateTime createdDate;

    protected Review() {
    }

    public Review(String rating, String description) {
        this.rating = rating;
        this.description = description;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getRating() {
        return rating;
    }

    public void setRating(String rating) {
        this.rating = rating;
    }

    public Course getCourse() {
        return course;
    }

    public void setCourse(Course course) {
        this.course = course;
    }

    @Override
    public String toString() {
        return "Review{" +
                "id=" + id +
                ", rating='" + rating + '\'' +
                ", description='" + description + '\'' +
                '}';
    }
}

File: Spring-Boot/jpa-kotlin-example/src/main/kotlin/dev/vitorvidal/jpakotlinexample/entity/Review.kt
package dev.vitorvidal.jpakotlinexample.entity

import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.time.LocalDateTime
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id

@Entity
class Review {
    @Id
    @GeneratedValue
    var id: Long? = null

    @Column
    var rating: String? = null

    @Column
    var description: String? = null

    @UpdateTimestamp
    val lastUpdatedDate: LocalDateTime? = null

    @CreationTimestamp
    val createdDate: LocalDateTime? = null

    protected constructor()
    constructor(description: String?) {
        this.description = description
    }

    override fun toString(): String {
        return "Review{" +
                "id=" + id +
                ", rating='" + rating + '\'' +
                ", description='" + description + '\'' +
                '}'
    }
}
